### PR TITLE
Upgrade prettier to 2.x

### PR DESCRIPTION
**Pull request by mroderick**
This PR upgrades `prettier` to latest and removes local configuration, adopting `prettier` default options.

**This breaks compatibility with IE11 and Edge Legacy, and is thus a `MAJOR` version release**

Also:
* upgrade `eslint-config-prettier`
* upgrade `eslint-plugin-prettier`
* run `prettier` on all JavaScript files
    ```sh
    $(npm bin)/prettier -w -u .
    ```

#### Purpose

We are exploring dropping IE11 and other legacy browsers. This PR demonstrates what the first step would look like.

#### Solution  - optional

* Upgrade `prettier`, `eslint-config-prettier` and `eslint-plugin-prettier` to latest
* Remove `.prettierrc`, accepting the default options
* Re-format all JavaScript files using `prettier`
* Configure editors via `.editorconfig`

#### How to verify - mandatory

1. Check out this branch
1. `npm install`
1. `npm run lint`
1. Observe that there are no warnings or errors

#### Checklist for author

- [x] `npm run lint` passes
- [x] References to standard library functions are [cached](https://github.com/sinonjs/sinon/pull/1523).

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -8,13 +8,8 @@ charset = utf-8
 end_of_line = lf
 insert_final_newline = true
 indent_style = space
-indent_size = 4
-trim_trailing_whitespace = true
-
-# Matches the exact files either package.json or .travis.yml
-[{package.json, .travis.yml}]
-indent_style = space
 indent_size = 2
```
Comment (mantoni) [inline .editorconfig:17]: Isn't this also a default config?

**Review by mantoni — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (mroderick) [reply .editorconfig:17]: In `EditorConfig`?

I couldn't find anything at https://editorconfig.org that shows that it has a default.

Comment (mantoni) [reply .editorconfig:17]: Ah, no. I confused it with the prettier defaults. All good.

**Comment by mroderick** [comment]
This has been published as `@sinonjs/referee@7.0.0`

**Pull request merged**